### PR TITLE
FIX: month start with 0 with Tests

### PR DESCRIPTION
**Pull request by james-ff**
This PR fixes month start with 0 and some other initial props. 
- Will deffinitely need review before being accepted. 

Some test cases have been constructed but commented out with TODO questions as to whether or not they should be used.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ dist/ui-coverage/
 npm-debug.log
 .publish
 .vscode
+package-lock.json
```
[the remark this answers is not in the document]
Comment (james-ff) [reply .gitignore]: absolutely

File: src/tests/DateRangePicker.spec.js [hunk 2]
```diff
@@ -180,6 +180,156 @@ describe('The DateRangePicker component', function () {
...
+          this.useDocumentRenderer({
+            selectionType: 'range',
+            value: null,
+            initialFromValue: true,
+          });
+          expect(this.renderedComponent.state.year).toBe(moment().year());
+          expect(this.renderedComponent.state.month).toBe(moment().month());
+        });
+
+        // TODO: Should this test exist? It currently causes errors as moment not expected
```
[the remark this answers is not in the document]
Comment (james-ff) [reply src/tests/DateRangePicker.spec.js]: Correct, have removed as this is not a desirable value input for that selection type, keeping tests focused on "happy case" inputs

File: src/tests/DateRangePicker.spec.js [hunk 3]
```diff
@@ -180,6 +180,156 @@ describe('The DateRangePicker component', function () {
...
+          this.useDocumentRenderer({
+            selectionType: 'single',
+            value: value,
+            initialFromValue: true,
+          });
+          expect(this.renderedComponent.state.year).toBe(value.year());
+          expect(this.renderedComponent.state.month).toBe(value.month());
+        });
+
+        // TODO: Should this test exist? It currently causes errors as momentRange not expected
```
[the remark this answers is not in the document]
Comment (james-ff) [reply src/tests/DateRangePicker.spec.js]: Correct, have removed as this is not a desirable value input for that selection type, keeping tests focused on "happy case" inputs

File: src/tests/DateRangePicker.spec.js [hunk 4]
```diff
@@ -180,6 +180,156 @@ describe('The DateRangePicker component', function () {
...
+        });
+
+        it('this.state set to value if value is a moment', function () {
+          var value = moment('2003 01 01', 'YYYY MM DD');
+          this.useDocumentRenderer({
+            selectionType: 'single',
+            value: value,
+            initialFromValue: true,
+          });
+          expect(this.renderedComponent.state.year).toBe(value.year());
```
[the remark this answers is not in the document]
Comment (james-ff) [reply src/tests/DateRangePicker.spec.js]: have done as such

File: src/tests/DateRangePicker.spec.js [hunk 5]
```diff
@@ -180,6 +180,156 @@ describe('The DateRangePicker component', function () {
...
+
+        it('this.state.month set to initialMonth if initialMonth is a minimum integer', function () {
+          this.useDocumentRenderer({
+            initialYear: 2000,
+            initialMonth: 0,
+          });
+          expect(this.renderedComponent.state.month).toBe(0);
+        });
+
+        // TODO: Should this test exist?
```
[the remark this answers is not in the document]
Comment (james-ff) [reply src/tests/DateRangePicker.spec.js]: done

File: src/tests/DateRangePicker.spec.js [hunk 6]
```diff
@@ -180,6 +180,156 @@ describe('The DateRangePicker component', function () {
...
+
+        it('this.state.month set to initialMonth if initialMonth is a maximum integer', function () {
+          this.useDocumentRenderer({
+            initialYear: 2000,
+            initialMonth: 11,
+          });
+          expect(this.renderedComponent.state.month).toBe(11);
+        });
+
+        // TODO: Should this test exist?
```
[the remark this answers is not in the document]
Comment (james-ff) [reply src/tests/DateRangePicker.spec.js]: done

**Comment by james-ff** [comment]
@AlanFoster Have made adjustments

**Comment by AlanFoster** [comment]
Awesome; Thanks! :+1:

**Pull request merged**